### PR TITLE
Sev fixes

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #
Closes #
Closes #

# Changes proposed in this pull request:
- Fixes issues with request loops and simplifies the code
- Fixes issues with no update packages after login
-

I made this a draft because I hope some other people like @Mekle001 could give this a quick test.
Some of the cycle code was sus, and Im not sure why we had it or if we needed it.

[ x ] Yes I tested this before submitting it.
[ ] I also did a multiplayer test.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java [hunk 1]
```diff
@@ -231,17 +225,17 @@ protected boolean createsCraftingCycle(
...
-     * Check if a building can craft a certain stack.
+     * Check if a building can craft a certain recipe.
      *
      * @param building       the building to check in.
-     * @param stackPredicate predicate used to check if a building knows the recipe.
+     * @param recipe         the recipe to check.
      * @return true if so.
      */
-    public abstract boolean canBuildingCraftStack(@NotNull final AbstractBuilding building, Predicate<ItemStack> stackPredicate);
+    public abstract boolean canBuildingCraftStack(@NotNull final AbstractBuilding building, IRecipeStorage recipe);
```
Comment (uecasm) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java]: This is more `canBuildingCraftRecipe` now, not `canBuildingCraftStack`.

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java [hunk 2]
```diff
@@ -132,36 +133,43 @@ public boolean canResolveRequest(@NotNull final IRequestManager manager, final I
...
 
-        if (createsCraftingCycle(manager, request, request))
+        if (createsCraftingCycle(manager, request, request.getRequest(), request))
         {
             return false;
         }
 
-        // As long as we're not resolving food, fast resolve
-        if(!(request.getRequest() instanceof Food))
+        final boolean isFood = !(request.getRequest() instanceof Food);
```
Comment (uecasm) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java]: `isFood == true` means it's **not** a food request?

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java]: Good point

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java (hunk 2 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java]: Good catch

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java [hunk 3]
```diff
@@ -145,13 +144,13 @@ public boolean canResolveForBuilding(@NotNull final IRequestManager manager, @No
...
 
-        final boolean isFood = !(request.getRequest() instanceof Food);
+        final boolean isFood = request.getRequest() instanceof Food;
         for (final ICraftingBuildingModule module : building.getModules(ICraftingBuildingModule.class))
         {
             final IRecipeStorage recipe = module.getFirstRecipe(itemStack -> request.getRequest().matches(itemStack));
 
             // If this building is resolving a generic food request, then only allow it to resolve non-smeltables.
-            if (recipe != null && (!isFood || recipe.getIntermediate() != Blocks.FURNACE) && canBuildingCraftStack(building, recipe))
+            if (recipe != null && (!isFood || recipe.getIntermediate() != Blocks.FURNACE) && canBuildingCraftRecipe(building, recipe))
```
Comment (uecasm) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java:153]: Is this logic still correct with the non-inverted `isFood`?  Or was it wrong before?

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java:153]: Yes

**Review by someaddons — APPROVED** [review]

**Pull request merged**